### PR TITLE
refactor: add package-json script naming convention and clean up scripts and workflows

### PR DESCRIPTION
**Pull request by oliverschuerch**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -41,72 +43,79 @@ The Web Content Accessibility Guidelines are a recommendation for designing acce
...
 We are happy to receive your input. You can submit your issues to our [GitHub repository](https://github.com/swisspost/design-system/issues). If you're rather looking for help, don't hesitate to open a discussion on [GitHub discussions](https://github.com/swisspost/design-system/discussions).
 
 When you're planning to work on bigger changes, please reach out to someone from the core team to plan your change.
 
+## Script name conventions
+
+Whenever we add new scripts to the package.json file, we follow the instructions below.
+
+- A script name is all lowercase.
+- A script name uses only letters, numbers and semi-colons.
```
Comment (gfellerph) [inline CONTRIBUTING.md:57]:
```suggestion
- A script name uses only letters, dashes and colons.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,72 +43,79 @@ The Web Content Accessibility Guidelines are a recommendation for designing acce
...
+- A script name short and descriptive.
```
Comment (gfellerph) [inline CONTRIBUTING.md:58]:
```suggestion
- A script name is short and descriptive.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -41,72 +43,79 @@ The Web Content Accessibility Guidelines are a recommendation for designing acce
...
+- A default script name persist of one word only (e.g. `lint`).
+- A non default script name is post-fix which describes it's special characteristica (e.g. `lint:fix`).
```
Comment (gfellerph) [inline CONTRIBUTING.md:60]:
```suggestion
- A non default script name is postfixed with it's characteristics (e.g. `lint:fix`).
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -41,72 +43,79 @@ The Web Content Accessibility Guidelines are a recommendation for designing acce
...
+- A non default script name for a specific tool, is pre-fixed with the toolname (e.g. `storybook:serve`, `storybook:bulid`, etc.).
```
Comment (gfellerph) [inline CONTRIBUTING.md:61]:
```suggestion
- A non default script name for a specific tool is prefixed with the tool name (e.g. `storybook:serve`, `storybook:bulid`, etc.).
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -41,72 +43,79 @@ The Web Content Accessibility Guidelines are a recommendation for designing acce
...
+- A non default script name can contain more than one semi-colon (e.g. `lint:fix:dry`, `stencil:test:watch`, etc.).
```
Comment (gfellerph) [inline CONTRIBUTING.md:62]:
```suggestion
- A non default script name can contain more than one colon (e.g. `lint:fix:dry`, `stencil:test:watch`, etc.).
```

**Review by gfellerph — CHANGES_REQUESTED** [review]

**Review by gfellerph — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 6]
```diff
@@ -18,12 +18,12 @@ jobs:
           fetch-depth: 0
 
       - name: Setup
-        uses: swisspost/design-system/.github/actions/setup-pnpm@main
+        uses: swisspost/design-system/.github/actions/setup-pnpm@refactor/549-refactor-packagejson-scripts
```
Comment (gfellerph) [inline .github/workflows/test.yaml:21]:
```suggestion
        uses: swisspost/design-system/.github/actions/setup-pnpm@main
```

**Review by gfellerph — CHANGES_REQUESTED** [review]
Is 82119b1d0e02fa57964041f485e0f6bc1d6f0d60 really necessary? GitHub Actions should be detected by pnpm as a CI environment.

**Comment by oliverschuerch** [comment]
@gfellerph your absolutly right. Github Workflows should be detected as CI and so the --frozen-lockfile flag is automatically set to true. By including the flag in the Workflows, I just wanted to make ist absolutly clear.
If you think it's redundant, I can remove it again.

**Review by gfellerph — APPROVED** [review]

**Pull request merged**